bitcoin/src/psbt/error.rs

```diff
@@ -80,6 +80,18 @@ pub enum Error {
     NegativeFee,
     /// Integer overflow in fee calculation
     FeeOverflow,
+    /// Parsing error indicating invalid public keys
+    InvalidPublicKey,
+    /// Parsing error indicating invalid xonly public keys
+    InvalidXOnlyPublicKey,
+    /// Parsing error indicating an invalid signature
+    InvalidSignature(&'static str),
+    /// Parsing error indicating invalid control block
+    InvalidControlBlock,
+    /// Parsing error indicating invalid leaf version
+    InvalidLeafVersion,
+    /// Parsing error indicating a Taproot error
+    TaprootError(&'static str),
```

Comment:
For `InvalidPublicKey` and `InvalidSignature` I think we should hold the `secp256k1::Error`.

Ideally, we'd hold the underlying error for the other variants too, though I understand that holding an `io::Error` would force us to remove some `#derive`s which might be controversial.

Also, after this PR are we able to get rid of the `ParseFailed` variant?

Reply:
Also `TaprootError` does not need the `Error` suffix.